### ISSUE TITLE
how can add extra audio streams codecs to my files before they get uploaded to gdrive ?

### ISSUE THREAD

**Issue by aniel300** [Unmanic/unmanic#226]
am trying to add extra audio streams to my files with both 7.1, 5.1 and 2.0 respectively (aac,ac3,eac3,truehd,dts to aac,ac3,eac3,truehd,dts) how do i go about doing this, i don't know what am doing. any help will be appreciated, thank you.

**Comment by Josh5** [comment]
5.1 to 7.1 seems like a waste of time.

Do you have an FFmpeg command that you would like to run? Or have you not managed to get that far yet?

**Comment by Josh5** [comment]
I am intending on writing a plugin to manage cloning stereo (2.0) streams from surround sound channels.

Will that suite your needs?

**Comment by aniel300** [comment]
yes but remember I also would like whatever original audio it comes with 7.1 or 5.1  (aac, ac3, eac3, truehd, dts) to be also replicated in (aac, ac3, eac3, truehd, dts) 7.1 or 5.1 respectively as well the 2.0 ones. my question is am going overkill and is it unnecessary to have that many codecs ?

**Comment by Josh5** [comment]
In my opinion it is. But what I would do with Unmanic may be completely different to what you might do

**Comment by aniel300** [comment]
so what codecs and channels would u have ? and are u gonna be able to help me or create the plugin ?

**Comment by Josh5** [comment]
If I had a 7.1 or 5.1 stream already, I would keep them as is and create a stereo 2.0 clone of that stream. I would pick aac for the stereo clone.

If you add my development repo, there is a plugin there that can do this.
https://github.com/Josh5/unmanic-plugins

**Comment by aniel300** [comment]
umm the reason am in pursuit of this taks is because rokus stick as an example does not like eac3 codec and am trying to eliminate transcoding, i will add the repo, however i think i still would like to do what am after

**Comment by aniel300** [comment]
after taking a look at this i don't think it is possible to aac, ac3, eac3 to truehd, dts however for what i need i think i should be fine with aac, ac3 and eac3.

**Comment by aniel300** [comment]
any update in regard to the taks i would like to do ?

**Comment by Josh5** [comment]
Did you try the plugin in the Dev repo?

**Comment by aniel300** [comment]
how can i run this command by its own ffmpeg -i "/media/merged/working/completed/movies/1.mkv" -map 0:v -c:v copy -map 0:1 -c:a:0 copy -map 0:a:0 -c:a:1 aac -map 0:a:0 -c:a:2 ac3 -map 0:a:0 -c:a:3 aac -ac:a:3 2 -map 0:s? "/media/merged/working/completed/movies/2.mkv"

**Comment by aniel300** [comment]
i need dynamic plugin that can ignore the whatever current codec is present

**Comment by Josh5** [comment]
This is not something I can do at this time. Feel free to try and write your own plugin for this. But at this time my plate is full.
If you dont mind waiting for a few months I may eventually get time to do this for you.
